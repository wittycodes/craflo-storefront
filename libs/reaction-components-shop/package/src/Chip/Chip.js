import React from "react";
import PropTypes from "prop-types";
import { Chip as MuiChip, makeStyles } from "@material-ui/core";
import clsx from "clsx";

const useStyles = makeStyles((theme) => ({
  success: {
    color: theme.palette.colors.coolGrey500,
    backgroundColor: theme.palette.colors.forestGreen100
  },
  info: {
    color: theme.palette.colors.coolGrey500,
    backgroundColor: theme.palette.colors.darkBlue100
  },
  danger: {
    color: theme.palette.colors.coolGrey500,
    backgroundColor: theme.palette.colors.red100
  },
  colorPrimaryError: {
    color: theme.palette.primary.contrastText,
    backgroundColor: theme.palette.colors.red
  },
  outlinedPrimaryError: {
    color: theme.palette.colors.red,
    border: `1px solid ${theme.palette.colors.red}`
  }
}));

/**
 * @name Chip
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */
const Chip = React.forwardRef(function Chip(props, ref) {
  const { color, variant, ...otherProps } = props;
  const classes = useStyles();

  const colorVariants = clsx({
    [classes.success]: color === "success",
    [classes.info]: color === "info",
    [classes.danger]: color === "danger"
  });

  let errorClasses = {};
  let errorColorProp = {};
  if (color === "error") {
    errorClasses = {
      colorPrimary: clsx({ [classes.colorPrimaryError]: variant === "default" }),
      outlinedPrimary: clsx({ [classes.outlinedPrimaryError]: variant === "outlined" })
    };
    errorColorProp = { color: "primary" };
  }

  let colorProp = {};
  // Only add props accepted by MUI Chip
  if (["default", "primary", "secondary"].includes(color)) {
    colorProp = { color };
  }

  return (
    <MuiChip
      {...colorProp}
      classes={{
        ...errorClasses,
        root: colorVariants
      }}
      variant={variant}
      {...errorColorProp}
      ref={ref}
      // eslint-disable-next-line react/jsx-indent-props
      {...otherProps}
    />
  );
});

Chip.propTypes = {
  /**
   * CSS Classes
   */
  classes: PropTypes.object,
  /**
   * The color of the component
   */
  color: PropTypes.oneOf(["default", "primary", "secondary", "success", "info", "danger", "error"]),
  /**
   * The variant to use
   */
  variant: PropTypes.oneOf(["default", "outlined"])
};

Chip.defaultProps = {
  color: "primary",
  variant: "default"
};

export default Chip;
