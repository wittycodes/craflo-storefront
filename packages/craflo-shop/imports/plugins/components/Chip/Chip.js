"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("core-js/modules/es7.object.get-own-property-descriptors");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.keys");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

require("core-js/modules/es7.array.includes");

require("core-js/modules/es6.string.includes");

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@material-ui/core");

var _clsx4 = _interopRequireDefault(require("clsx"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var useStyles = (0, _core.makeStyles)(function (theme) {
  return {
    success: {
      color: theme.palette.colors.coolGrey500,
      backgroundColor: theme.palette.colors.forestGreen100
    },
    info: {
      color: theme.palette.colors.coolGrey500,
      backgroundColor: theme.palette.colors.darkBlue100
    },
    danger: {
      color: theme.palette.colors.coolGrey500,
      backgroundColor: theme.palette.colors.red100
    },
    colorPrimaryError: {
      color: theme.palette.primary.contrastText,
      backgroundColor: theme.palette.colors.red
    },
    outlinedPrimaryError: {
      color: theme.palette.colors.red,
      border: "1px solid ".concat(theme.palette.colors.red)
    }
  };
});
/**
 * @name Chip
 * @param {Object} props Component props
 * @returns {React.Component} returns a React component
 */

var Chip = _react.default.forwardRef(function Chip(props, ref) {
  var _clsx;

  var color = props.color,
      variant = props.variant,
      otherProps = (0, _objectWithoutProperties2.default)(props, ["color", "variant"]);
  var classes = useStyles();
  var colorVariants = (0, _clsx4.default)((_clsx = {}, (0, _defineProperty2.default)(_clsx, classes.success, color === "success"), (0, _defineProperty2.default)(_clsx, classes.info, color === "info"), (0, _defineProperty2.default)(_clsx, classes.danger, color === "danger"), _clsx));
  var errorClasses = {};
  var errorColorProp = {};

  if (color === "error") {
    errorClasses = {
      colorPrimary: (0, _clsx4.default)((0, _defineProperty2.default)({}, classes.colorPrimaryError, variant === "default")),
      outlinedPrimary: (0, _clsx4.default)((0, _defineProperty2.default)({}, classes.outlinedPrimaryError, variant === "outlined"))
    };
    errorColorProp = {
      color: "primary"
    };
  }

  var colorProp = {}; // Only add props accepted by MUI Chip

  if (["default", "primary", "secondary"].includes(color)) {
    colorProp = {
      color: color
    };
  }

  return /*#__PURE__*/_react.default.createElement(_core.Chip, (0, _extends2.default)({}, colorProp, {
    classes: _objectSpread(_objectSpread({}, errorClasses), {}, {
      root: colorVariants
    }),
    variant: variant
  }, errorColorProp, {
    ref: ref // eslint-disable-next-line react/jsx-indent-props

  }, otherProps));
});

Chip.propTypes = {
  /**
   * CSS Classes
   */
  classes: _propTypes.default.object,

  /**
   * The color of the component
   */
  color: _propTypes.default.oneOf(["default", "primary", "secondary", "success", "info", "danger", "error"]),

  /**
   * The variant to use
   */
  variant: _propTypes.default.oneOf(["default", "outlined"])
};
Chip.defaultProps = {
  color: "primary",
  variant: "default"
};
var _default = Chip;
exports.default = _default;