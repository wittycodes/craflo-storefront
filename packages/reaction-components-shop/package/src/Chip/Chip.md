### Overview

Chips are compact visual elements that are used to present information to an operator to communicate status or feedback on an action taken. Chips can be used to represent tags, badges, and other compact bits of information to a user.

[Chips](https://material.io/design/components/chips.html)Â allow users to enter information, make selections, filter content, or trigger actions.

The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information.

### Usage

<!-- Show all the variants/combos we use in Reaction Admin, without the code box > -->

```jsx noeditor
const onDelete = () => {
  console.log("stuff");
};
<div>
  <div style={{ display: "flex" }}>
    <div style={{ marginRight: "1rem" }}>
      <Chip color="error" label="status chip" />
    </div>
    <div style={{ marginRight: "1rem" }}>
      <Chip label="deletable chip" color="primary" onDelete={onDelete} />
    </div>
    <div style={{ marginRight: "1rem" }}>
      <Chip
        label="small deletable chip for multiselect"
        color="secondary"
        size="small"
        onDelete={onDelete}
      />
    </div>
  </div>
</div>;
```

#### Types

<!-- Show all Types of the component used in Reaction Admin -->

##### Deletable chip

Use a Deletable chip to allow users to remove an entity, like removing a Tag from a filter input. To create a Deletable chip, pass a Delete function to `onDelete`, and use the `default` variant with the `primary` color.

```jsx
function ChipsArray() {
  const [chipData, setChipData] = React.useState([
    { key: 0, label: "swim-2019.csv" },
    { key: 1, label: "womens-2019.csv" },
    { key: 2, label: "mens-2019.csv" }
  ]);

  const handleDelete = chipToDelete => () => {
    setChipData(chips => chips.filter(chip => chip.key !== chipToDelete.key));
  };

  return (
    <div>
      {chipData.map(data => {
        return (
          <Chip
            color="primary"
            key={data.key}
            label={data.label}
            onDelete={handleDelete(data)}
            style={{ marginRight: "4px" }}
          />
        );
      })}
    </div>
  );
}

<ChipsArray />;
```

##### Deletable chip for MultiSelects

Use a Deletable chip in components a MultiSelect to allow users to remove entities within another action. To create a Deletable chip for a Select or MultiSelect, use the `default` variant with the `secondary` color and `small` size props, along with p assing an `onDelete` function.

```jsx
function ChipsArray() {
  const [chipData, setChipData] = React.useState([
    { key: 0, label: "womens" },
    { key: 1, label: "mens" },
    { key: 2, label: "fall-winter-2019" }
  ]);

  const handleDelete = chipToDelete => () => {
    setChipData(chips => chips.filter(chip => chip.key !== chipToDelete.key));
  };

  return (
    <div>
      {chipData.map(data => {
        return (
          <Chip
            color="secondary"
            size="small"
            key={data.key}
            label={data.label}
            onDelete={handleDelete(data)}
            style={{ marginRight: "4px" }}
          />
        );
      })}
    </div>
  );
}

<ChipsArray />;
```

##### Error chip

The Error chip is used to indicate an error status, such as when an order has been cancelled.

```jsx
<Chip color="error" label="Order Cancelled" />
```

##### All Chip variants

These are all the possible chip variants.

```jsx
function ChipsArray() {
  const handleDelete = chipToDelete => () => {
    alert("You clicked the delete button");
  };

  const [chipData, setChipData] = React.useState([
    [
      {
        label: "primary-default",
        color: "primary",
        variant: "default"
      },
      { label: "primary-outlined", color: "primary", variant: "outlined" },
      {
        label: "primary-deletable",
        color: "primary",
        key: 0,
        onDelete: handleDelete
      },
      {
        label: "primary-deletable-outlined",
        color: "primary",
        variant: "outlined",
        onDelete: handleDelete
      }
    ],
    [
      { label: "secondary-default", color: "secondary", variant: "default" },
      { label: "secondary-outlined", color: "secondary", variant: "outlined" },
      {
        label: "secondary-deletable",
        color: "secondary",
        onDelete: handleDelete
      },
      {
        label: "secondary-deletable-outlined",
        color: "secondary",
        variant: "outlined",
        onDelete: handleDelete
      }
    ],
    [
      { label: "error-default", color: "error" },
      { label: "error-outlined", color: "error", variant: "outlined" },
      {
        label: "error-deletable-outlined",
        color: "error",
        variant: "outlined",
        onDelete: handleDelete
      }
    ],
    [
      { label: "success-default", color: "success" },
      { label: "success-outlined", color: "success", variant: "outlined" },
      { label: "success-deletable", color: "success", onDelete: handleDelete },
      {
        label: "success-deletable-outlined",
        color: "success",
        variant: "outlined",
        onDelete: handleDelete
      }
    ],
    [
      { label: "info-default", color: "info" },
      {
        label: "info-outlined",
        color: "info",
        variant: "outlined"
      },
      {
        label: "info-deletable",
        color: "info",
        onDelete: handleDelete
      },
      {
        label: "info-deletable-outlined",
        color: "info",
        variant: "outlined",
        onDelete: handleDelete
      }
    ],
    [
      { label: "danger-default", color: "danger" },
      { label: "danger-outlined", color: "danger", variant: "outlined" },
      {
        label: "danger-deletable",
        color: "danger",
        onDelete: handleDelete
      },
      {
        label: "danger-deletable-outlined",
        color: "danger",
        variant: "outlined",
        onDelete: handleDelete
      }
    ]
  ]);

  return (
    <div>
      {chipData.map((chipRow, index) => {
        return (
          <div key={index} style={{ marginBottom: "8px" }}>
            {chipRow.map((data, chipIndex) => {
              return (
                <Chip
                  color="secondary"
                  {...data}
                  key={chipIndex}
                  label={data.label}
                  onDelete={data.onDelete && data.onDelete(data)}
                  style={{ marginRight: "4px" }}
                />
              );
            })}
          </div>
        );
      })}
    </div>
  );
}
<ChipsArray />;
```
