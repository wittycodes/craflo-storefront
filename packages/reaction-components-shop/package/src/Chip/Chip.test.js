import React from "react";
import { render } from "../tests/index.js";
import Chip from "./Chip";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<Chip />);
  expect(asFragment()).toMatchSnapshot();
});

test("deletable chip snapshot", () => {
  const onDelete = () => { };
  const { asFragment } = render(<Chip color="primary" variant="default" onDelete={onDelete} />);
  expect(asFragment()).toMatchSnapshot();
});

test("deletable chip in small, secondary sizesnapshot", () => {
  const onDelete = () => { };
  const { asFragment } = render(<Chip color="secondary" variant="default" onDelete={onDelete} size="small" />);
  expect(asFragment()).toMatchSnapshot();
});


test("error chip snapshot", () => {
  const { asFragment } = render(<Chip color="error"/>);
  expect(asFragment()).toMatchSnapshot();
});
