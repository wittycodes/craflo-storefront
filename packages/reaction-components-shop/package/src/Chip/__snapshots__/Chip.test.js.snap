// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorPrimary"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</DocumentFragment>
`;

exports[`deletable chip in small, secondary sizesnapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorSecondary MuiChip-deletableColorSecondary MuiChip-sizeSmall MuiChip-deletable"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelSmall"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorSecondary MuiChip-deleteIconSmall"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`deletable chip snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`error chip snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-colorPrimary makeStyles-colorPrimaryError-127"
  >
    <span
      class="MuiChip-label"
    />
  </div>
</DocumentFragment>
`;
